replica/table.cc

```diff
@@ -1627,14 +1633,12 @@ compaction_group::compaction_group(table& t, size_t group_id, dht::token_range t
 }
 
 future<> compaction_group::stop() noexcept {
-    try {
-        // FIXME: make memtable_list::flush() noexcept too.
-        return _memtables->flush().finally([this] {
-            return _t._compaction_manager.remove(as_table_state());
-        });
-    } catch (...) {
-        return current_exception_as_future<>();
+    if (_async_gate.is_closed()) {
+        co_return;
     }
+    co_await _async_gate.close();
+    co_await flush();
```

Comment:
@raphaelsc although flush was made noexcept, it might return a failed future
and then `co_await`ing will throw.

This doesn't fix the FIXME above.